src/main/java/io/github/mzmine/modules/featuredetection/msmsdetection/MsMsDetectionTask.java

```diff
@@ -0,0 +1,196 @@
+            @Nonnull Double intensityTolerance, @Nullable String nameSuffix) {
+        this.rawDataFile = rawDataFile;
+        this.dataStore = dataStore;
+        this.mzTolerance = mzTolerance;
+        this.rtTolerance = rtTolerance;
+        this.intensityTolerance = intensityTolerance;
+        this.nameSuffix = nameSuffix;
+        this.title = title;
+        this.message = message;
+
+        // New feature filter task
+        msMsDetectionMethod = new MsMsDetectionMethod(rawDataFile,
+                scanSelection, dataStore, mzTolerance, rtTolerance,
+                intensityTolerance);
+
+        // Targeted detection method
+        targetedDetectionMethod = new TargetedDetectionMethod(null, null, null,
+                null, null, null, null);
+
+        // Chromatogram to feature table method
+        chromatogramToFeatureTableMethod = new ChromatogramToFeatureTableMethod(
+                null, null, null);
+
+        refreshStatus();
+
+        EventHandler<WorkerStateEvent> cancelEvent = new EventHandler<WorkerStateEvent>() {
+            @Override
+            public void handle(WorkerStateEvent workerEvent) {
+                msMsDetectionMethod.cancel();
+                targetedDetectionMethod.cancel();
+                chromatogramToFeatureTableMethod.cancel();
+            }
+        };
+
+        setOnCancelled(cancelEvent);
+    }
+
+    @Override
+    public void refreshStatus() {
+
+        // Progress
+        Float finishedPercent = 0f;
+
+        final Float method1Percent = msMsDetectionMethod
+                .getFinishedPercentage();
+        if (method1Percent != null)
+            finishedPercent = method1Percent * 0.1f;
+
+        final Float method2Percent = targetedDetectionMethod
+                .getFinishedPercentage();
+        if (method2Percent != null)
+            finishedPercent = finishedPercent + method2Percent * 0.8f;
+
+        final Float method3Percent = chromatogramToFeatureTableMethod
+                .getFinishedPercentage();
+        if (method3Percent != null)
+            finishedPercent = finishedPercent + method3Percent * 0.1f;
+
+        updateProgress(finishedPercent.doubleValue(), 1.0);
+
+        // Title and message
+        updateTitle(title);
+        updateMessage(message);
+    }
+
+    @Override
+    protected Object call() throws Exception {
+        try {
+            List<IonAnnotation> ionAnnotations = (List<IonAnnotation>) msMsDetectionMethod
+                    .execute();
+
+            // Run the targeted feature detection module
+            this.targetedDetectionMethod = new TargetedDetectionMethod(
+                    ionAnnotations, rawDataFile, dataStore, mzTolerance,
+                    rtTolerance, intensityTolerance, minHeight);
+
+            // Run method
+            try {
+                List<Chromatogram> detectedChromatograms = (List<Chromatogram>) targetedDetectionMethod
+                        .execute();
+

```

Comment:
I'm unable to execute the three methods after each other. The msMsDetectionMethod runs fine and I get a list of IonAnnotations. But after that I get a NULL error in TargetedDetectionMethod (l. 184). Any suggestions about what is wrong?